sphinx_gallery/gen_rst.py

```diff
@@ -512,11 +511,8 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot') or not gallery_conf['plot_gallery']:
-        convert_func = dict(code=codestr2rst, text=text2string)
-        for blabel, bcontent in script_blocks:
-            example_rst += convert_func[blabel](bcontent) + '\n'
-    else:
+    filename_pattern = gallery_conf.get('filename_pattern')
```

Comment:
> umm ... I think this was based on a comment by @Titan-C . We have a default somewhere in gen_gallery.rst

I have to say I didn't really understand his comment at the time. At the moment, filename_pattern is not defined in conf.py for any of the projects using sphinx-gallery. That means they'll get an error when they update sphinx-gallery. I guess we can either:
* have a default in this `.get` call so that everything works as before.
* raise a meaningful error if filename_pattern is not set saying that filename_pattern needs to be set in conf.py and what to set it to if they just want things to work as before.